File: app/src/main/java/de/bitb/astroskop/controller/ControllerCallback.java
package de.bitb.astroskop.controller;

import java.util.List;

public class ControllerCallback<O> {
    public void onSuccess(List<O> response) {
        //implement if needed
    }

    public void onSuccess(O response) {
        //implement if needed
    }

    public void onSuccess() {
        //implement if needed
    }

    public void onFailure() {
        //implement if needed
    }

}

File: app/src/main/java/de/bitb/astroskop/controller/ControllerCallback.kt
package de.bitb.astroskop.controller

class ControllerCallback<in O> {
    fun onSuccess(response: List<O>) {
        //implement if needed
    }

    fun onSuccess(response: O) {
        //implement if needed
    }

    fun onSuccess() {
        //implement if needed
    }

    fun onFailure() {
        //implement if needed
    }

}
